KifuSF/CreateDonationViewController.swift

```diff
@@ -0,0 +1,246 @@
+//
+//  CreateDonationViewController.swift
+//  KifuSF
+//
+//  Created by Alexandru Turcanu on 17/12/2018.
+//  Copyright © 2018 Alexandru Turcanu. All rights reserved.
+//
+
+import UIKit
+import LocationPicker
+
+class CreateDonationViewController: UIScrollableViewController {
+    //MARK: - Variables
+    private static let descriptionPlaceholder = "Additional Info about pick up address, hour, item, etc."
+    private let imageHelper = PhotoHelper()
+    private var userSelectedAProfileImage: Bool? = nil
+
+    private var pickupLocation: Location?
+
+    private let descriptorView = UIDescriptorView(defaultImageViewSize: .medium)
+    private let titleInputView = UIGroupView<UITextFieldContainer>(title: "Title",
+                                                                   contentView: UITextFieldContainer(returnKeyType: .next,
+                                                                                                     placeholder: "Keep it simple"))
+    private let descriptionInputView = UIGroupView<UITextView>(title: "Description",
+                                                               contentView: UITextView(forAutoLayout: ()))
+    private let pickUpAddressButton = UIAnimatedButton(backgroundColor: .kfInformative,
+                                                       andTitle: "Choose pick-up address")
+    private let errorLabel = UILabel(font: UIFont.preferredFont(forTextStyle: .footnote), textColor: .kfDestructive)
+
+    //MARK: - Lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureData()
+        configureStyling()
+        configureLayout()
+        configureGestures()
+        configureDelegates()
+
+        imageHelper.completionHandler = { [unowned self] (image) in
+            self.descriptorView.imageView.image = image
+            self.userSelectedAProfileImage = true
+        }
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        NotificationCenter.default.addObserver(self,selector: #selector(keyboardWillShow(_:)),
+                                               name: .UIKeyboardWillShow, object: nil)
+
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillHide(_:)),
+                                               name: .UIKeyboardWillHide, object: nil)
```

Comment:
I have this helper that can handles the observing of the keyboard events and uses delegates to notify when the keyboard comes and goes, if you're interested to add it to the project here it is:

```swift
import Foundation
import UIKit.UIWindow

protocol KeyboardStackDelegate: class {
    func keyboard(_ keyboard: KeyboardStack, didChangeTo newHeight: CGFloat)
}

class KeyboardStack: NSObject {
    
    weak var delegate: KeyboardStackDelegate?
    
    private var keyboardHieght: CGFloat = 0
    
    override init() {
        super.init()
        
        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillChange(notification:)), name: NSNotification.Name.UIKeyboardWillShow, object: nil)
        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillChange(notification:)), name: NSNotification.Name.UIKeyboardWillHide, object: nil)
        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillChange(notification:)), name: NSNotification.Name.UIKeyboardWillChangeFrame, object: nil)
    }
    
    deinit {
        NotificationCenter.default.removeObserver(self, name: NSNotification.Name.UIKeyboardWillShow, object: nil)
        NotificationCenter.default.removeObserver(self, name: NSNotification.Name.UIKeyboardWillHide, object: nil)
        NotificationCenter.default.removeObserver(self, name: NSNotification.Name.UIKeyboardWillChangeFrame, object: nil)
    }
    
    @objc private func keyboardWillChange(notification: Notification) {
        guard let keyboardRect = (notification.userInfo?[UIKeyboardFrameEndUserInfoKey] as? NSValue)?.cgRectValue else {
            return
        }
        
        if notification.name == Notification.Name.UIKeyboardWillShow ||
            notification.name == Notification.Name.UIKeyboardWillChangeFrame {
            self.delegate?.keyboard(self, didChangeTo: keyboardRect.height + 64)
        } else {
            self.delegate?.keyboard(self, didChangeTo: 0)
        }
    }
}
```

Reply:
Integrated something similar to this :]